versions.tf

```diff
@@ -1,7 +1,14 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# SET TERRAFORM AND PROVIDER REQUIREMENTS FOR RUNNING THIS MODULE
+# ----------------------------------------------------------------------------------------------------------------------
+
 terraform {
   required_version = ">= 0.12.20, < 2.0"
 
   required_providers {
-    aws = ">= 2.0, < 4.0"
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 2.0, < 4.0"
+    }
```

Comment:
is this supported by terraform 0.12?

Reply:
Terraform v0.12.26 will accept syntax like the above but will understand it in the same way as the following v0.12-style syntax:

terraform {
  required_providers {
    aws = "~> 1.0"
  }
}

Reply:
> is this supported by terraform 0.12?

ah shit.. will change